Mapify/SceneInitializers/Railway/TrackSetup.cs

```diff
@@ -84,41 +88,190 @@ private void TestConnections()
+
+            SetupStalk(prefabClone, customSwitch.GetJointPoint());
+            DestroyPrefabTracks(prefabClone);
+            CreateSwitchTracks(customSwitch, prefabClone, inJunction);
+
+            foreach (var track in customSwitch.GetTracks())
+            {
+                track.gameObject.SetActive(true);
+            }
+        }
+
+        private static void DestroyPrefabTracks(GameObject prefabClone)
+        {
+            // must be destroyed immediately to prevent:
+            // "Junction 'in_junction' doesn't have track '[track diverging]' assigned"
+            // from RailManager.TestConnections
+            Object.DestroyImmediate(prefabClone.FindChildByName(Switch.THROUGH_TRACK_NAME));
+            Object.DestroyImmediate(prefabClone.FindChildByName(Switch.DIVERGING_TRACK_NAME));
+        }
+
+        private static void CreateSwitchTracks(CustomSwitch customSwitch, GameObject prefabClone, Junction switchJunction)
+        {
+            var railTracksInSwitch = CreateRailTracks(
+                customSwitch.GetTracks(), false
+            );
+
+            if (!railTracksInSwitch.Any())
+            {
+                Mapify.LogError($"{nameof(CreateCustomSwitches)}: {nameof(railTracksInSwitch)} is empty");
+                return;
+            }
+
+            switchJunction.outBranches = new List<Junction.Branch>();
+
+            foreach (var trackInSwitch in railTracksInSwitch)
+            {
+                trackInSwitch.transform.SetParent(prefabClone.transform, true);
+
+                trackInSwitch.inBranch = new Junction.Branch();
+                trackInSwitch.inBranch.track = null;
+                trackInSwitch.inJunction = switchJunction;
+
+                switchJunction.outBranches.Add(new Junction.Branch(trackInSwitch, true));
+            }
+
+            //track before the switch
+            switchJunction.inBranch = switchJunction.FindClosestBranch(railTracksInSwitch[0].curve[0].transform.position);
+
+            //connect the track before the switch to the switch
+            if (switchJunction.inBranch.first)
             {
-                Transform swTransform = sw.transform;
-                VanillaAsset vanillaAsset = sw.GetComponent<VanillaObject>().asset;
+                switchJunction.inBranch.track.inJunction = switchJunction;
+            }
+            else
+            {
+                switchJunction.inBranch.track.outJunction = switchJunction;
+            }
+
+            if (switchJunction.inBranch == null)
+            {
+                Mapify.LogError($"{nameof(CreateSwitchTracks)}: inBranch is null");
+            }
+        }
+
+        private static void SetupStalk(GameObject prefabClone, BezierPoint joinPoint)
+        {
+            //visual objects
+            var graphical = prefabClone.FindChildByName("Graphical").transform;
+            string[] toDelete = {"ballast", "anchors", "sleepers", "rails_static", "rails_moving"};
```

Comment:
This should be `static readonly` so it doesn't allocate a new array for every call.

Reply:
I'll make it a class member instead of a local variable